### PR TITLE
Chef 13 Updates

### PR DESCRIPTION
**Pull request by ramereth**
This provides a variety of fixes including by not limited to:
- Move database and mysql recipes into gwm_test cookbook
- Move all code into a single recipe
- Switch to using attributes in .kitchen.yml
- Convert from ServerSpec to InSpec and also improve tests
- Fix installation issues
- Add support for SSLCertificateChainFile

For more detailed information, please look at the commits.

This PR is currently missing ChefSpec tests which @jcorme will be adding later.

### CODE REVIEW

**Comment by ramereth** [comment]
[Chef 13: ganeti_webmgr](https://trello.com/c/SdooL0HS/880-chef-13-ganetiwebmgr)

**Comment by TravisWhitehead** [comment]
For the `default suite`, converge fails on the `execute[install_gwm]` resource.
The `mysql` suite is failing no `chef_gem[mysql]`.

**Comment by jcorme** [comment]
@ramereth @TravisWhitehead ChefSpec tests added.

File: spec/unit/recipes/default_spec.rb [hunk 1]
```diff
@@ -4,12 +4,241 @@
...
+      end
+
+      it do
+        expect(chef_run).to sync_git('/opt/ganeti_webmgr_src').with(
+          repository: 'https://github.com/osuosl/ganeti_webmgr',
+          revision: 'develop',
+          user: 'ganeti_webmgr',
+          group: 'ganeti_webmgr'
+        )
+      end
```
Comment (ramereth) [inline spec/unit/recipes/default_spec.rb]: Can you please add a check for this resource notifying the execute resource for installing GWM?

File: libraries/matchers.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    )
+  end
+
+  def enable_web_app(resource_name)
+    ChefSpec::Matchers::ResourceMatcher.new(
+      :web_app,
+      :enable,
+      resource_name
+    )
+  end
```
[the remark this answers is not in the document]
Comment (jcorme) [reply libraries/matchers.rb]: Yes, there's a web_app resource in the default recipe that I wanted to test, but for some reason, ChefSpec doesn't see it.

**Comment by ramereth** [comment]
!bump major gwm

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a major-level version bump to v1.0.0.  Have a nice day!

**Pull request merged**